assets/src/vo/currency.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * External imports
+ */
+import { isEmpty, isString, isNumber, isBoolean } from 'lodash';
+import { Exception, CURRENCY_CONFIG } from '@eventespresso/eejs';
+import warning from 'warning';
+
+/**
+ * A value object representing currency values
+ */
+export class Currency {
+	/**
+	 * The code identifying the currency (eg. 'USD')
+	 * @type {string}
+	 */
+	code = '';
+
+	/**
+	 * The singular label for the currency (eg. 'Dollar');
+	 * @type {string}
+	 */
+	singularLabel = '';
+
+	/**
+	 * The plural label for the currency (eg. 'Dollars');
+	 * @type {string}
+	 */
+	pluralLabel = '';
```

Comment:
I'm still not sure whether I'll have the Currency Vo require usage, but I'm going to create a Label value object as a part of the work I'm doing for the hardcoded pretty status labels in another branch (turns out its a good thing to implement there to save having to pick from arrays).

Reply:
right... I've thought many times about using a Label VO in our PHP code, but it would have to replace existing code which causes all kinds of issues.
Since this stuff is all starting out from scratch I felt I should mention it before it is too late to change. The main benefits of using a VO for this is it standardizes our usage of labels for things, and can potentially make some process easier.

That said, not using a VO for labels doesn't complicate things much so not going to push for this.